app/controllers/teams_controller.rb

```diff
@@ -1,23 +1,35 @@
 class TeamsController < ApplicationController
   before_action :authenticate_user!
+  before_action :ensure_valid_team_id, except: [:create, :index, :new_team]
   before_action :authenticate_promote_leader_params, only: [:promote_to_leader]
+  before_action :ensure_leader, only: [:update]
+
+  def ensure_valid_team_id
+    @team = Team.find(params[:id])
+    if @team == nil
+      redirect_to home_path
+    end
+  end
 
   def authenticate_promote_leader_params
-    @team = Team.find(params[:team_id])
     @new_leader = User.find(params[:user_id])
     if !current_user.is_leader_of(@team) || !(@team.members.include? @new_leader)
       redirect_to team_path(@team)
     end
   end
 
+  def ensure_leader
+    if !current_user.is_leader_of(@team)
+      redirect_to home_path
+    end
+  end
+
   def index
   	@message = "Hello, #{current_user.first}!"
     @teams = Team.all
   end
 
   def show
-  	id = params[:id]
```

Comment:
Why the removal of show?

Reply:
because of the "before_action :ensure_valid_team_id" line